Gems/ROS2/Code/Source/Gripper/GripperActionServer.cpp

```diff
@@ -0,0 +1,142 @@
+    {
+        actionServer.reset();
+        AZ::TickBus::Handler::BusDisconnect();
+    }
+
+    void GripperActionServer::Reflect(AZ::ReflectContext* context)
+    {
+        if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
+        {
+            serialize->Class<GripperActionServer, AZ::Component>()
+                ->Field("ActionServerName", &GripperActionServer::m_gripperActionServerName)
+                ->Version(1);
+
+            if (AZ::EditContext* ec = serialize->GetEditContext())
+            {
+                ec->Class<GripperActionServer>("GripperActionServer", "GripperActionServer")
+                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "GripperActionServer")
+                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC("Game"))
+                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2")
+                    ->DataElement(
+                        AZ::Edit::UIHandlers::EntityId,
+                        &GripperActionServer::m_gripperActionServerName,
+                        "Gripper Action Server",
+                        "Gripper Action Server.");
+            }
+        }
+    }
+
+    bool GripperActionServer::IsGoalActiveState() const
+    {
+        return m_goalHandle->is_active() || m_goalHandle->is_executing() || m_goalHandle->is_canceling();
+    }
+
+    void GripperActionServer::OnTick(float delta, AZ::ScriptTimePoint timePoint)
+    {
+        if (m_goalHandle && IsGoalActiveState())
+        {
+            auto feedback = std::make_shared<GripperCommand::Feedback>();
+            float position = 0.0;
+            float effort = 0.0;
+            bool isDone = false;
+
+            GripperRequestBus::EventResult(position, GetEntityId(), &GripperRequestBus::Events::GetGripperPosition);
+            GripperRequestBus::EventResult(effort, GetEntityId(), &GripperRequestBus::Events::GetGripperEffort);
+            GripperRequestBus::EventResult(isDone, GetEntityId(), &GripperRequestBus::Events::IsGripperReachedGoal);
+
+            if (isDone)
+            {
+                AZ_Printf("GripperActionServer::OnTick", "GripperActionServer::OnTick: Gripper reached goal!");
+                auto result = std::make_shared<GripperCommand::Result>();
+                result->position = position;
+                result->effort = effort;
+                result->reached_goal = true;
+                m_goalHandle->succeed(result);
+            }
+            else
+            {
+                feedback->position = position;
+                feedback->position = effort;
+                feedback->reached_goal = false;
+                m_goalHandle->publish_feedback(feedback);
+            }
+        }
+    }
+
+    rclcpp_action::GoalResponse GripperActionServer::handleGoal(
+        const rclcpp_action::GoalUUID& uuid, std::shared_ptr<const GripperCommand::Goal> goal)
+    {
+        (void)uuid;
```

Comment:
use `[[maybe_unused]]` instead
```suggestion
    rclcpp_action::GoalResponse GripperActionServer::handleGoal(
        [[maybe_unused]] const rclcpp_action::GoalUUID& uuid, std::shared_ptr<const GripperCommand::Goal> goal)
    {
```